contracts/modules/CoveredCalls/Core/ETH/CoveredCallsEth.sol

```diff
@@ -0,0 +1,274 @@
+
+        // Store the specified expiration
+        call.expiry = _expiry;
+
+        emit CallCreated(_tokenContract, _tokenId, call);
+    }
+
+    ///                                                          ///
+    ///                        CANCEL CALL                       ///
+    ///                                                          ///
+
+    /// @notice Emitted when a covered call option is canceled
+    /// @param tokenContract The ERC-721 token address of the canceled call option
+    /// @param tokenId The ERC-721 token id of the canceled call option
+    /// @param call The metadata of the canceled call option
+    event CallCanceled(address tokenContract, uint256 tokenId, Call call);
+
+    /// @notice Cancels a call option that has not yet been purchased
+    /// @param _tokenContract The ERC-721 token address
+    /// @param _tokenId The ERC-721 token id
+    function cancelCall(address _tokenContract, uint256 _tokenId) external {
+        // Get the option for the specified token
+        Call memory call = callForNFT[_tokenContract][_tokenId];
+
+        // Ensure the option has not been purchased
+        require(call.buyer == address(0), "PURCHASED");
+
+        // Ensure the caller is the seller or a new token owner
+        require(msg.sender == call.seller || msg.sender == IERC721(_tokenContract).ownerOf(_tokenId), "ONLY_SELLER_OR_TOKEN_OWNER");
+
+        emit CallCanceled(_tokenContract, _tokenId, call);
+
+        // Remove the option from storage
+        delete callForNFT[_tokenContract][_tokenId];
+    }
+
+    ///                                                          ///
+    ///                         BUY CALL                         ///
+    ///                                                          ///
+
+    /// @notice Emitted when a covered call option is purchased
+    /// @param tokenContract The ERC-721 token address of the purchased call option
+    /// @param tokenId The ERC-721 token id of the purchased call option
+    /// @param call The metadata of the purchased call option
+    event CallPurchased(address tokenContract, uint256 tokenId, Call call);
+
+    /// @notice Purchases a call option for an NFT
+    /// @param _tokenContract The address of the ERC-721 token
+    /// @param _tokenId The ERC-721 token id
+    /// @param _strike The strike price of the option
+    function buyCall(
+        address _tokenContract,
+        uint256 _tokenId,
+        uint256 _strike
+    ) external payable nonReentrant {
+        // Get the option for the specified token
+        Call storage call = callForNFT[_tokenContract][_tokenId];
+
+        // Ensure the option has not been purchased
+        require(call.buyer == address(0), "INVALID_PURCHASE");
+
+        // Ensure the option has not expired
+        require(call.expiry > block.timestamp, "INVALID_CALL");
+
+        // Ensure the specified strike matches the call strike
+        require(call.strike == _strike, "MUST_MATCH_STRIKE");
```

Comment:
why even pass strike if they have to match? 🤔

Reply:
its to prevent a buyer from getting rugged on a strike price they didn't intend to purchase the option for

there'd be a race condition upon a pending buy where the seller could cancel the call and create a new one with the same premium but higher strike price